Documentation/building/linux-instructions.md

```diff
@@ -12,9 +12,11 @@ Build using Docker
 
 Install Docker, see https://docs.docker.com/install/
 
-Building using Docker will require that you choose the correct image for your environment. Note that the OS is strictly speaking not extremely important, for example if you are on Ubuntu 18.04 and build using the Ubuntu 16.04 x64 image there should be no issues. The target architecture is more important, as building arm32 using the x64 image will not work, there will be missing rootfs components required by the build. See [Docker Images](#Docker-Images) for more information on choosing an image to build with.
+Note that you should not mix docker builds and non docker builds because some of the files are shared. It's best to just make a new clone of the repo for each docker build and make a separate clone for non docker builds. 
```

Comment:
I think there are also issues that come from cmake's cache. I believe that it should be suggested instead of cloning again to cleaning the enlistment. For example, I believe running the following will hit an issue where the <coreclr>/bin/ folder would need to be deleted before a successful build happens.

`docker run -rm -i -v home/coreclr:/coreclr -w /coreclr ./build.sh arm64 checked`

At this point, because, cross was not passed you will have x64 binaries under <coreclr>/bin/Product/Linux.arm64.Checked.

Which will have generated an incorrect cmake cache under bin/... Which means when you want to run the correct command with `cross` passed, you will get an error because the cmake generated cache will not match.

`docker run -rm -i -v home/coreclr:/coreclr -w /coreclr ./build.sh arm64 checked`

This should be easily fixed by running clean.sh (or if you do not want to nuke your repo): rm -r bin/Product/Linux.arm64.Checked && rm -r bin/obj/Linux.arm64.Checked

Reply:
I dunno and since I prefer to spend as little time on this as possible it seemed the easiest to simply make a new clone - making a new clone takes very little time and works fine so that was the approach I took. you are welcome to document a different way that doesn't require a new clone, of course. I just haven't tried it myself.